### PR TITLE
[gluster] remove only dump files

### PR DESCRIPTION
**Pull request by jjansky1**
Removes only dump files and leaving
other files as .socket or sock.

Resolves: #2152

Signed-off-by: Jan Jansky <jjansky@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [x] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: sos/report/plugins/gluster.py [hunk 1]
```diff
@@ -41,15 +41,16 @@ def wait_for_statedump(self, name_dir):
...
-            os.rmdir(self.statedump_dir)
-            os.unlink('/tmp/glusterdump.options')
-        except OSError:
-            pass
+        if self.get_option("dump"):
+            if not os.path.exists(self.statedump_dir):
+                return
+            try:
+                for name in os.listdir(self.statedump_dir):
+                    if 'dump' in name:
```
Comment (bmr-cymru) [inline sos/report/plugins/gluster.py:49]: Please use the full pattern we discussed on IRC: `*.dump.*`: we know that all Gluster dump files conform to this pattern, so it is the most strict we can make it without adding cumbersome steps like PID parsing and checking the timestamp portion falls in a recent window.

File: sos/report/plugins/gluster.py [hunk 2]
```diff
@@ -41,15 +41,16 @@ def wait_for_statedump(self, name_dir):
...
+                        os.remove(os.path.join(self.statedump_dir, name))
+                os.unlink('/tmp/glusterdump.options')
```
Comment (bmr-cymru) [inline sos/report/plugins/gluster.py:51]: This is a left-over from the patch that introduced the regression:

```diff
commit 7b485f02c39b0d8502c224dd90c27024951948dc
Author: Jake Hunsaker <jhunsake@redhat.com>
Date:   Wed Apr 17 17:00:03 2019 -0400

    [gluster] Add volume, state information and dump glusterd processes
[...]
@@ -40,17 +40,6 @@ class Gluster(Plugin, RedHatPlugin):
         fp.close()
         return out
 
-    def make_preparations(self, name_dir):
-        try:
-            os.mkdir(name_dir)
-        except OSError:
-            pass
-        fp = open('/tmp/glusterdump.options', 'w')
-        data = 'path=' + name_dir + '\n'
-        fp.write(data)
-        fp.write('all=yes')
-        fp.close()
-
```

All the code that generated this file was already removed so we should get rid of this as well.

File: sos/report/plugins/gluster.py (hunk 1 above)
Comment (sanjurakonde) [reply sos/report/plugins/gluster.py:49]: I suggest we use .*dump.[0-9]*, statedump files end with dump.epoch time

Comment (sanjurakonde) [reply sos/report/plugins/gluster.py:49]: it should be .\*dump.[0-9]\*

File: sos/report/plugins/gluster.py (hunk 2 above)
Comment (jjansky1) [reply sos/report/plugins/gluster.py]: os.unlink('/tmp/glusterdump.options') removed

File: sos/report/plugins/gluster.py (hunk 1 above)
Comment (jjansky1) [reply sos/report/plugins/gluster.py]: .*dump.[0-9]* not working due to glob syntax used 

                for name in glob.glob(self.statedump_dir + '/*.dump.[0-9]*'):
                    os.remove(name)

File: sos/report/plugins/gluster.py [hunk 3]
```diff
@@ -41,15 +41,14 @@ def wait_for_statedump(self, name_dir):
...
-                os.remove(os.path.join(self.statedump_dir, dirs))
-            os.rmdir(self.statedump_dir)
-            os.unlink('/tmp/glusterdump.options')
-        except OSError:
-            pass
+        if self.get_option("dump"):
+            if not os.path.exists(self.statedump_dir):
+                return
+            try:
+                for name in glob.glob(self.statedump_dir + '/*.dump.[0-9]*'):
```
Comment (sanjurakonde) [inline sos/report/plugins/gluster.py:48]: not sure if I'm missing something, don't we need to have '.' after the '/'?

Comment (sanjurakonde) [inline sos/report/plugins/gluster.py:48]: not sure if I'm missing something, don't we need to have '.' after the '/'?

Comment (TurboTurtle) [reply sos/report/plugins/gluster.py:48]: Threw me off at first too, but `glob` uses UNIX-style regex, not python-style.

Comment (bmr-cymru) [reply sos/report/plugins/gluster.py:48]: No - these are not "dotfiles". They start with a string like "glusterdump" or a mount point and end with  ".dump.$TIMESTAMP":

```# ls /var/run/gluster/*.dump.[0-9]*
/var/run/gluster/glusterdump.1452.dump.1594652912   /var/run/gluster/glusterdump.24959.dump.1594652913  /var/run/gluster/glusterdump.8205.dump.1594813835   /var/run/gluster/mnt-data2-2.24719.dump.1594652912
/var/run/gluster/glusterdump.1452.dump.1594652913   /var/run/gluster/glusterdump.24959.dump.1594813835  /var/run/gluster/glusterdump.8245.dump.1594652912   /var/run/gluster/mnt-data2-2.24719.dump.1594652913
/var/run/gluster/glusterdump.1452.dump.1594813835   /var/run/gluster/glusterdump.24961.dump.1594652912  /var/run/gluster/glusterdump.8245.dump.1594652913   /var/run/gluster/mnt-data2-2.24719.dump.1594813835
/var/run/gluster/glusterdump.1454.dump.1594652912   /var/run/gluster/glusterdump.24961.dump.1594652913  /var/run/gluster/glusterdump.8245.dump.1594813835   /var/run/gluster/var-lib-glusterd-ss_brick.14293.dump.1594652912
/var/run/gluster/glusterdump.1454.dump.1594652913   /var/run/gluster/glusterdump.24961.dump.1594813835  /var/run/gluster/mnt-data1-1.24697.dump.1594652912  /var/run/gluster/var-lib-glusterd-ss_brick.14293.dump.1594652913
/var/run/gluster/glusterdump.1454.dump.1594813835   /var/run/gluster/glusterdump.8205.dump.1594652912   /var/run/gluster/mnt-data1-1.24697.dump.1594652913  /var/run/gluster/var-lib-glusterd-ss_brick.14293.dump.1594813835
/var/run/gluster/glusterdump.24959.dump.1594652912  /var/run/gluster/glusterdump.8205.dump.1594652913   /var/run/gluster/mnt-data1-1.24697.dump.1594813835
```

Comment (bmr-cymru) [reply sos/report/plugins/gluster.py:48]: No, glob uses ["globs"](https://en.wikipedia.org/wiki/Glob_(programming)) :-D - they are a totally different (although superficially similar) string pattern matching scheme that originates in the Bourne shell.

Comment (sanjurakonde) [reply sos/report/plugins/gluster.py:48]: I don't have knowledge of how glob works. if it is something not missed by mistake, I'm good.

**Pull request closed without merging**